### PR TITLE
AlertingNG: Add webhook notification channel

### PR DESCRIPTION
**Pull request by codesome**
**What this PR does / why we need it**:

It has all the fields from Prometheus Alertmanager webhook payload and some fields from existing Grafana webhook notification channel.

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/email.go [hunk 1]
```diff
@@ -74,8 +72,8 @@ func (en *EmailNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool,
 				"CommonLabels":      data.CommonLabels,
 				"CommonAnnotations": data.CommonAnnotations,
 				"ExternalURL":       data.ExternalURL,
-				"RuleUrl":           path.Join(en.appURL, "/alerting/list"),
-				"AlertPageUrl":      path.Join(en.appURL, "/alerting/list?alertState=firing&view=state"),
+				"RuleUrl":           path.Join(en.externalUrl.String(), "/alerting/list"),
```
Comment (davidmparrott) [inline pkg/services/ngalert/notifier/channels/email.go:75]: It looks like this is breaking the test case now

**Comment by codesome** [comment]
```
Linting back-end code, edition: "OSS"
--
3 | pkg/services/ngalert/notifier/channels/telegram.go:6: File is not `goimports`-ed (goimports)
4 | "fmt"
5 | golangci-lint failed: exit status 1
```

It is in right order :thinking:

File: pkg/services/ngalert/notifier/alertmanager.go [hunk 2]
```diff
@@ -355,23 +360,21 @@ func (am *Alertmanager) buildReceiverIntegrations(receiver *apimodels.PostableAp
...
 			err error
 		)
-		externalURL, err := url.Parse(am.Settings.AppURL)
-		if err != nil {
-			return nil, err
-		}
 		switch r.Type {
 		case "email":
-			n, err = channels.NewEmailNotifier(cfg, externalURL, am.Settings.AppURL)
+			n, err = channels.NewEmailNotifier(cfg, tmpl.ExternalURL)
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/alertmanager.go:365]:
```suggestion
			n, err = channels.NewEmailNotifier(cfg, tmpl.ExternalURL) // email notifier already has a default template.
```

File: pkg/services/ngalert/notifier/channels/webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+			if c.expInitError != nil {
+				require.Error(t, err)
+				require.Equal(t, c.expInitError.Error(), err.Error())
+				return
+			}
+			require.NoError(t, err)
+
+			body := ""
+			bus.AddHandlerCtx("test", func(ctx context.Context, webhook *models.SendWebhookSync) error {
+				body = webhook.Body
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/channels/webhook_test.go:173]: Should be include the other methods to make sure they correct defaults are set at least? URL, User, Password, HttpMethod, etc.

File: pkg/services/ngalert/notifier/channels/webhook_test.go [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+	tmpl.ExternalURL = externalURL
+
+	cases := []struct {
+		name         string
+		settings     string
+		alerts       []*types.Alert
+		expMsg       *webhookMessage
+		expInitError error
+		expMsgError  error
+	}{
```
Comment (gotjosh) [inline pkg/services/ngalert/notifier/channels/webhook_test.go:36]: Could we add a case where we truncate the alerts too? 😄

**Review by gotjosh — APPROVED** [review]
LGTM, my comments are mostly nits.

File: pkg/services/ngalert/notifier/channels/webhook_test.go (hunk 3 above)
Comment (codesome) [reply pkg/services/ngalert/notifier/channels/webhook_test.go:173]: Added those custom settings to existing test cases

File: pkg/services/ngalert/notifier/channels/webhook_test.go (hunk 4 above)
Comment (codesome) [reply pkg/services/ngalert/notifier/channels/webhook_test.go:36]: Modified current case with truncation

**Pull request merged**